tests/model/test_rnn.py

```diff
@@ -35,13 +36,24 @@ def rnn_types(draw) -> st.SearchStrategy[RNNType]:
 
 @st.composite
 def rnns(
-    draw, return_kwargs: bool = False, input_size: Optional[int] = None
+    draw,
+    return_kwargs: bool = False,
+    input_size: Optional[int] = None,
+    hard_lstm: Optional[bool] = None,
 ) -> Union[
-    st.SearchStrategy[RNN], st.SearchStrategy[Tuple[RNN, Dict]],
+    st.SearchStrategy[Union[RNN, HardLSTM]],
+    st.SearchStrategy[Tuple[Union[RNN, HardLSTM], Dict]],
 ]:
     """Returns a SearchStrategy for RNN."""
-    kwargs = {}
-    kwargs["rnn_type"] = draw(rnn_types())
+    kwargs: Dict = {}
+    if hard_lstm is not True:
+        kwargs["rnn_type"] = draw(rnn_types())
+
+    if hard_lstm is None and kwargs["rnn_type"] == RNNType.LSTM:
+        hard_lstm = st.booleans()
+        if hard_lstm:
+            del kwargs["rnn_type"]  # this is not an arg for HardLSTM
```

Comment:
Why is `hard_lstm` an optional parameter? It seems like it should be non-optional and this becomes:

```suggestion
    if hard_lstm:
        assume(kwargs["rnn_type"] == RNNType.LSTM)
        del kwargs["rnn_type"]  # this is not an arg for HardLSTM
```

Reply:
It is optional so that when the arg isn't provided (for `test_rnn_forward_pass_correct_shapes_returned()`) hypothesis tests both `RNN` and `HardLSTM` at the same time. 

In terms of the ordering of:
``` 
if not hard_lstm:
...
if hard_lstm is None and kwargs["rnn_type"] == RNNType.LSTM:
    hard_lstm = st.booleans()
```
my thinking here is that it would be weirder to choose whether an LSTM should be hard or not before deciding the RNN type.